tgui/packages/tgui/interfaces/Filteriffic.js

```diff
@@ -0,0 +1,306 @@
+        mr={0.5} />
+      <Input
+        value={value}
+        width="90px"
+        onInput={(e, value) => act('transition_filter_value', {
+          name: filterName,
+          new_data: {
+            [name]: value,
+          },
+        })} />
+    </Fragment>
+  );
+};
+
+const FilterIconEntry = (props, context) => {
+  const { value, filterName } = props;
+  const { act } = useBackend(context);
+  return (
+    <Fragment>
+      <Button
+        icon="pencil-alt"
+        onClick={() => act('modify_icon_value', {
+          name: filterName,
+        })} />
+      <Box inline ml={1}>
+        {value}
+      </Box>
+    </Fragment>
+  );
+};
+
+const FilterFlagsEntry = (props, context) => {
+  const { name, value, filterName, filterType } = props;
+  const { act, data } = useBackend(context);
+
+  const filterInfo = data.filter_info;
+  const flags = filterInfo[filterType]['flags'];
+  return (
+    map((bitField, flagName) => (
+      <Button.Checkbox
+        checked={value & bitField}
+        content={flagName}
+        onClick={() => act('modify_filter_value', {
+          name: filterName,
+          new_data: {
+            [name]: value ^ bitField,
+          },
+        })} />
+    ))(flags)
+  );
+};
+
+const FilterDataEntry = (props, context) => {
+  const { name, value, hasValue, filterName } = props;
+
+  const filterEntryTypes = {
+    int: <FilterIntegerEntry {...props} />,
+    float: <FilterFloatEntry {...props} />,
+    string: <FilterTextEntry {...props} />,
+    color: <FilterColorEntry {...props} />,
+    icon: <FilterIconEntry {...props} />,
+    flags: <FilterFlagsEntry {...props} />,
+  };
+
+  const filterEntryMap = {
+    x: 'float',
+    y: 'float',
+    icon: 'icon',
+    render_source: 'string',
+    flags: 'flags',
+    size: 'float',
+    color: 'color',
+    offset: 'float',
+    radius: 'float',
+    falloff: 'float',
+    density: 'int',
+    threshold: 'float',
+    factor: 'float',
+    repeat: 'int',
+  };
+
+  return (
+    <LabeledList.Item label={name}>
+      {filterEntryTypes[filterEntryMap[name]] || "Not Found (This is an error)"}
+      {' '}
+      {!hasValue && <Box inline color="average">(Default)</Box>}
+    </LabeledList.Item>
+  );
+};
+
+const FilterEntry = (props, context) => {
+  const { act, data } = useBackend(context);
+  const { name, filterDataEntry } = props;
+  const { type, priority, ...restOfProps } = filterDataEntry;
+
+  const filterDefaults = data["filter_info"];
+
+  const targetFilterPossibleKeys = Object.keys(filterDefaults[type]['defaults']);
+
+  return (
+    <Collapsible
+      title={name + " (" + type + ")"}
+      buttons={(
+        <Fragment>
+          <Button
+            icon="chevron-up"
+            onClick={() => act('increase_priority', { name: name })} />
+          <Box inline mr={1}>
+            {priority}
+          </Box>
+          <Button
+            icon="chevron-down"
+            onClick={() => act('decrease_priority', { name: name })} />
```

Comment:
Would it be possible to use a NumberInput here?

Reply:
ngl I don't know if I care enough to properly get that working now. The idea was originally that the arrows determine the direction the priority moves in but then I realized they're not actually sorted.

Reply:
Mostly for the looks.